WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailActivity.kt

```diff
@@ -13,11 +13,16 @@ class HistoryDetailActivity : LocaleAwareActivity() {
         const val KEY_HISTORY_DETAIL_FRAGMENT = "history_detail_fragment"
     }
 
+    private var binding: HistoryDetailActivityBinding? = null
```

Comment:
Warning (⚠️): When using `ViewBinding` within `activities`, it is not required to have it nullable and then nulling is out during `onDestroy()`. This is only needed for `Fragments` (see [here](https://developer.android.com/topic/libraries/view-binding#fragments)). For `Activities` it is suggested to use `lateinit` instead, see [here](https://developer.android.com/topic/libraries/view-binding#activities).

PS: Actually you did exactly that for `SuggestionActivity`. Was there something different here?
PSS: I found that the same comment applies for the `DomainRegistrationActivity` and `PostsListActivity` classes, which might have the wrong configuration. @planarvoid can you verify my assumptions so that we can correct that and avoiding replicating by example this unnecessary `ViewBinding` null related configuration for `Activities` in the future?

Reply:
@ParaskP7  - same comment as above. Let's use this opportunity to agree on the direction forward.

Reply:
👋 @zwarm !

True, but I do believe, unless I am totally wrong, that in this case, on the `Activities`, we definitely shouldn't use nullable `ViewBinding` fields when needed, and instead use `lateinit` for the same purpose, as suggested in the official documentation. I also pinged Vojta to double verify on that, so that we get rid of the remaining such inconsistencies, and in order for us to move on towards the recommended approach going forward. 🙏

Reply:
FWIW I agree that using lateinit and keeping the binding as non-nullable in activities seems like the best approach (and also recommended). 

P.S. Vojta is AFK this week.

Reply:
@ParaskP7  Updated in d3f7f58

Reply:
👋 @zwarm !

LGTM 🙏